src/server.c

```diff
@@ -3396,6 +3407,16 @@ struct redisCommand *lookupCommandOrOriginal(sds name) {
 void propagate(struct redisCommand *cmd, int dbid, robj **argv, int argc,
                int flags)
 {
+    /* This should be unreachable while clients are paused, but modules or
+     * a multi-exec with client pause + write can reach here, so log a warning
+     * instead of a serverAssert() to record the misuse. */
+    if (!clientsArePaused()) {
+        serverLog(LL_WARNING, "Command '%s' propagated to replicas " 
```

Comment:
I can do that. I was torn about asserting specifically because people are using this to failover, it's probably one of the worst times to crash.

Reply:
I want some way to catch this in the test suite. 
We can maybe add some keyword that there tests search for in the logs. 

On the other hand, out philosophy so far was to crash when something unexpected happens.. We may cause someone some damage, but at least we'll learn about the problem, and fix it asap instead of keeping buggy code that can cause data corruption or inconsistencies. 

I'm not certain if this one falls under this philosophy, or we wanna be more careful here. But at the very least the warning should be bolder.

Reply:
@oranagra My thought was to implement a serverLog(LL_ERROR) that crashes during tests, but just reports an error in real use cases. Seem sufficient?

Reply:
I found a semi-ok work around.

Reply:
i'm not certain we have to support that. why would someone do that?
why wouldn't they do
```
MULTI
SET FOO BAR
CLIENT PAUSE
EXEC
```
seems more straight forward.

is there are real use case for that, or are we just concerned not to break something that users may already be using?

Reply:
Doing the set before the CLIENT PAUSE is more sensible, but it will still trigger the assert as long as CLIENT PAUSE is sent in a MULTI-BLOCK that propagates something. Since it's within a MULTI-BLOCK, the final exec will always propagate at the end of the block. Since client pause will now be active, that will trigger a crash. 

I don't think it's a likely case, but it seems possible someone might trigger it. I think the assert is really useful as you mentioned, to make sure we're being correct here.